libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioHelper.cs

```diff
@@ -61,31 +59,52 @@ public static CreateMessageOptions ActivityToTwilio(Activity activity, string tw
+        /// Writes the HttpResponse.
         /// </summary>
-        /// <param name="httpRequest">The HTTP request.</param>
-        /// <param name="validationUrl">Optional validation URL to override the automatically
-        /// generated URL signature used to validate incoming requests.</param>
-        /// <param name="authToken">The authentication token for the Twilio app.</param>
-        /// <returns>The activity object.</returns>
-        /// <seealso cref="TwilioAdapter.ProcessAsync(HttpRequest, HttpResponse, IBot, System.Threading.CancellationToken)"/>
-        /// <seealso cref="TwilioAdapterOptions.ValidationUrl"/>
-        public static async Task<Activity> RequestToActivity(HttpRequest httpRequest, Uri validationUrl, string authToken)
+        /// <param name="response">The httpResponse.</param>
+        /// <param name="code">The status code to be written.</param>
+        /// <param name="text">The text to be written.</param>
+        /// <param name="encoding">The encoding for the text.</param>
+        /// <param name="cancellationToken">A cancellation token for the task.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public static async Task WriteAsync(HttpResponse response, int code, string text, Encoding encoding, CancellationToken cancellationToken)
         {
-            if (httpRequest == null)
+            if (response == null)
             {
-                return null;
+                throw new ArgumentNullException(nameof(response));
             }
 
-            Dictionary<string, string> body;
-            using (var bodyStream = new StreamReader(httpRequest.Body))
+            if (text == null)
             {
-                body = QueryStringToDictionary(await bodyStream.ReadToEndAsync().ConfigureAwait(false));
+                throw new ArgumentNullException(nameof(text));
             }
 
-            ValidateRequest(httpRequest, body, validationUrl, authToken);
+            if (encoding == null)
+            {
+                throw new ArgumentNullException(nameof(encoding));
+            }
+
+            response.ContentType = "text/plain";
+            response.StatusCode = code;
+
+            var data = encoding.GetBytes(text);
 
-            var twilioMessage = JsonConvert.DeserializeObject<TwilioMessage>(JsonConvert.SerializeObject(body));
+            await response.Body.WriteAsync(data, 0, data.Length, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Creates a Bot Framework <see cref="Activity"/> from an HTTP request that contains a Twilio message.
+        /// </summary>
+        /// <param name="payload">The HTTP request.</param>
+        /// <returns>The activity object.</returns>
+        public static Activity PayloadToActivity(Dictionary<string, string> payload)
+        {
+            if (payload == null)
+            {
+                return null;
```

Comment:
As per the advice on the other adapters (confirmed with Chris / Gabo) please throw an exception here instead of returning null.